test/k8pshTest.cxx

```diff
@@ -0,0 +1,149 @@
+			return 1;
+		}
+		else if (arg == "2")
+		{
+			TEST_THAT(k8psh::Utilities::getEnvironmentVariable("PATH") == ".");
+			std::cout << std::cin.rdbuf();
+
+			return 2;
+		}
+		else if (arg == "3")
+		{
+			TEST_THAT(k8psh::Utilities::getEnvironmentVariable("PATH") != ".");
+			(void)std::fclose(stdin);
+			(void)std::fclose(stdout);
+			(void)std::fclose(stderr);
+			std::this_thread::sleep_for(std::chrono::milliseconds(250));
+
+			return 3;
+		}
+
+		return mainClient(argc, argv);
+	}
+	else if (basename != "k8pshTest")
+		return mainClient(argc, argv);
+
+	// Create configuration file
+	std::ofstream configFile((basename + ".conf").c_str());
+
+	configFile << "workingDirectory = ." << std::endl;
+	configFile << "[k8pshTest]" << std::endl;
+	configFile << "'0_" << basename << "' K8PSH_TEST_NAME= PATH= '" << executable << "' 0" << std::endl;
+	configFile << "'1_" << basename << "' K8PSH_TEST_NAME= =PATH= '" << executable << "' 1" << std::endl;
+	configFile << "'2_" << basename << "' K8PSH_TEST_NAME= ?PATH= '" << executable << "' 2" << std::endl;
+	configFile << "'3_" << basename << "' K8PSH_TEST_NAME= ?PATH= '" << executable << "' 3" << std::endl;
+	configFile.close();
+
+	// Start server
+	k8psh::Utilities::setEnvironmentVariable("K8PSH_CONFIG", basename + ".conf");
+	k8psh::Utilities::setEnvironmentVariable("K8PSH_DEBUG", "Main, Process");
+
+	std::thread([executable]{ (void)runCommand(executable + " --name k8pshTest --generate-local-executables --timeout 3000"); }).detach();
+	std::this_thread::sleep_for(std::chrono::milliseconds(250));
+	k8psh::Utilities::setEnvironmentVariable("K8PSH_DEBUG");
+
+	// Run test cases
+	TEST_THAT(k8psh::Utilities::setEnvironmentVariable("PATH", "."));
+	TEST_THAT(k8psh::Utilities::setEnvironmentVariable("K8PSH_TEST_NAME", "Test 0"));
+	TEST_THAT(runCommand("0_" + basename + " > test.out 2> test.err") == 0);
+	TEST_THAT(k8psh::Utilities::readFile("test.out") == static_cast<const std::ostringstream &>(std::ostringstream() << "Test 0").str());
+	TEST_THAT(k8psh::Utilities::readFile("test.err").empty());
+
+	TEST_THAT(k8psh::Utilities::setEnvironmentVariable("K8PSH_TEST_NAME", "Test 1"));
+	TEST_THAT(runCommand("1_" + basename + " a > test.out 2> test.err") == 1);
+	TEST_THAT(k8psh::Utilities::readFile("test.out").empty());
+	TEST_THAT(k8psh::Utilities::readFile("test.err") == static_cast<const std::ostringstream &>(std::ostringstream() << "Test 1, a").str());
+
+	TEST_THAT(runCommand("2_" + basename + " < test.err > test.out") == 2);
+	TEST_THAT(k8psh::Utilities::readFile("test.out") == static_cast<const std::ostringstream &>(std::ostringstream() << "Test 1, a").str());
+
+	TEST_THAT(k8psh::Utilities::setEnvironmentVariable("PATH", ".;.:."));
+	TEST_THAT(runCommand("3_" + basename + " < test.out > test3.out 2> test.err") == 3);
+	TEST_THAT(k8psh::Utilities::readFile("test3.out").empty());
+	TEST_THAT(k8psh::Utilities::readFile("test.err").empty());
+}
```

Comment:
Call `std::exit(0)`, so the test will complete quickly, without waiting for the server to terminate.